### PR TITLE
Fix werkzeug instance was created twice in Debug Mode (#2135)

### PR DESCRIPTION
**Pull request by asdf2014**
Fix werkzeug instance was created twice in Debug Mode (#2135)
```
(superset) [root@edeppreapp01 superset-0.15.4]# ./bin/python ./lib/python2.7/site-packages/superset/cli.py runserver -d -p 8888
/root/superset-0.15.4/lib/python2.7/site-packages/flask/exthook.py:71: ExtDeprecationWarning: Importing flask.ext.script is deprecated, use flask_script instead.
  .format(x=modname), ExtDeprecationWarning
/root/superset-0.15.4/lib/python2.7/site-packages/flask/exthook.py:71: ExtDeprecationWarning: Importing flask.ext.sqlalchemy is deprecated, use flask_sqlalchemy instead.
  .format(x=modname), ExtDeprecationWarning
/root/superset-0.15.4/lib/python2.7/site-packages/flask/exthook.py:71: ExtDeprecationWarning: Importing flask.ext.sqlalchemy._compat is deprecated, use flask_sqlalchemy._compat instead.
  .format(x=modname), ExtDeprecationWarning
/root/superset-0.15.4/lib/python2.7/site-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.
  warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "
/root/superset-0.15.4/lib/python2.7/site-packages/flask/exthook.py:71: ExtDeprecationWarning: Importing flask.ext.cache is deprecated, use flask_cache instead.
  .format(x=modname), ExtDeprecationWarning
2017-02-08 15:40:07,200:INFO:flask_appbuilder.base:Registering class MyIndexView on menu 
2017-02-08 15:40:07,205:INFO:flask_appbuilder.base:Registering class UtilView on menu 
2017-02-08 15:40:07,209:INFO:flask_appbuilder.base:Registering class LocaleView on menu 
2017-02-08 15:40:07,213:INFO:flask_appbuilder.base:Registering class ResetPasswordView on menu 
2017-02-08 15:40:07,230:INFO:flask_appbuilder.base:Registering class ResetMyPasswordView on menu 
2017-02-08 15:40:07,244:INFO:flask_appbuilder.base:Registering class UserInfoEditView on menu 
2017-02-08 15:40:07,258:INFO:flask_appbuilder.base:Registering class AuthDBView on menu 
2017-02-08 15:40:07,267:INFO:flask_appbuilder.base:Registering class UserDBModelView on menu List Users
2017-02-08 15:40:07,313:INFO:flask_appbuilder.base:Registering class RoleModelView on menu List Roles
2017-02-08 15:40:07,355:INFO:flask_appbuilder.base:Registering class UserStatsChartView on menu User's Statistics
2017-02-08 15:40:07,383:INFO:flask_appbuilder.base:Registering class PermissionModelView on menu Base Permissions
2017-02-08 15:40:07,418:INFO:flask_appbuilder.base:Registering class ViewMenuModelView on menu Views/Menus
2017-02-08 15:40:07,452:INFO:flask_appbuilder.base:Registering class PermissionViewModelView on menu Permission on Views/Menus
2017-02-08 15:40:08,112:INFO:flask_appbuilder.base:Registering class TableColumnInlineView on menu 
2017-02-08 15:40:08,158:INFO:flask_appbuilder.base:Registering class DruidColumnInlineView on menu 
2017-02-08 15:40:08,193:INFO:flask_appbuilder.base:Registering class SqlMetricInlineView on menu 
2017-02-08 15:40:08,223:INFO:flask_appbuilder.base:Registering class DruidMetricInlineView on menu 
2017-02-08 15:40:08,265:WARNING:flask_appbuilder.models.filters:Filter type not supported for column: password
2017-02-08 15:40:08,269:INFO:flask_appbuilder.base:Registering class DatabaseView on menu Databases
2017-02-08 15:40:08,308:WARNING:flask_appbuilder.models.filters:Filter type not supported for column: password
2017-02-08 15:40:08,311:INFO:flask_appbuilder.base:Registering class DatabaseAsync on menu 
2017-02-08 15:40:08,337:WARNING:flask_appbuilder.models.filters:Filter type not supported for column: password
2017-02-08 15:40:08,340:INFO:flask_appbuilder.base:Registering class DatabaseTablesAsync on menu 
2017-02-08 15:40:08,374:INFO:flask_appbuilder.base:Registering class TableModelView on menu Tables
2017-02-08 15:40:08,421:INFO:flask_appbuilder.base:Registering class AccessRequestsModelView on menu Access requests
2017-02-08 15:40:08,472:INFO:flask_appbuilder.base:Registering class DruidClusterModelView on menu Druid Clusters
2017-02-08 15:40:08,605:INFO:flask_appbuilder.base:Registering class SliceModelView on menu Slices
2017-02-08 15:40:08,640:INFO:flask_appbuilder.base:Registering class SliceAsync on menu 
2017-02-08 15:40:08,670:INFO:flask_appbuilder.base:Registering class SliceAddView on menu 
2017-02-08 15:40:08,699:INFO:flask_appbuilder.base:Registering class DashboardModelView on menu Dashboards
2017-02-08 15:40:08,737:INFO:flask_appbuilder.base:Registering class DashboardModelViewAsync on menu 
2017-02-08 15:40:08,767:INFO:flask_appbuilder.base:Registering class LogModelView on menu Action Log
2017-02-08 15:40:08,809:INFO:flask_appbuilder.base:Registering class QueryView on menu Queries
2017-02-08 15:40:08,851:INFO:flask_appbuilder.base:Registering class DruidDatasourceModelView on menu Druid Datasources
2017-02-08 15:40:08,891:INFO:flask_appbuilder.base:Registering class R on menu 
2017-02-08 15:40:08,896:INFO:flask_appbuilder.base:Registering class Superset on menu 
2017-02-08 15:40:08,986:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-08 15:40:09,026:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-08 15:40:09,121:INFO:werkzeug: * Running on http://0.0.0.0:8888/ (Press CTRL+C to quit)

```
(superset) [root@edeppreapp01 superset-0.15.4]# ps -ef | grep superset | grep -v grep
root     31522  1632 28 15:40 pts/0    00:00:03 ./bin/python ./lib/python2.7/site-packages/superset/cli.py runserver -d -p 8888

### CODE REVIEW

**Comment by mistercrunch** [comment]
Doesn't this prevent the debug server to auto-restart when files are modified?

**Comment by asdf2014** [comment]
I see. Could u give me a solution about the case that remote debug superset with pydevd in pycharm?

**Comment by asdf2014** [comment]
Just like i said in [https://github.com/airbnb/superset/issues/2135](https://github.com/airbnb/superset/issues/2135), we cannot remote debug with multi processes cuz the `reloader` function in werkzeug or gunicorn etc...

**Comment by mistercrunch** [comment]
Maybe make `reloader` as a switch in the `superset runserver` CLI command. Make sure the default is to reload as most people probably aren't using PyCharm

**Comment by asdf2014** [comment]
@mistercrunch Like this?
The type of the result from `config.get("FLASK_USE_RELOAD")` is bool, however the type of the result from  `@manager.option` is str, so i just convert them into str and compare with 'True'.
Now, Test results:
a) `(superset) [root@edeppreapp01 superset-0.15.4]# ./bin/python ./bin/superset runserver -d -p 8888`
/root/superset-0.15.4/lib/python2.7/site-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.
  warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "
2017-02-10 10:38:30,629:INFO:flask_appbuilder.base:Registering class MyIndexView on menu 
2017-02-10 10:38:30,634:INFO:flask_appbuilder.base:Registering class UtilView on menu 
// ...
2017-02-10 10:38:32,606:INFO:flask_appbuilder.base:Registering class Superset on menu 
2017-02-10 10:38:32,693:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-10 10:38:32,731:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-10 10:38:32,826:INFO:werkzeug: * Running on http://0.0.0.0:8888/ (Press CTRL+C to quit)
2017-02-10 10:38:32,827:INFO:werkzeug: * Restarting with stat
// ...
2017-02-10 10:38:35,491:INFO:flask_appbuilder.base:Registering class Superset on menu 
2017-02-10 10:38:35,578:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-10 10:38:35,617:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-10 10:38:35,712:WARNING:werkzeug: * Debugger is active!
2017-02-10 10:38:35,713:INFO:werkzeug: * Debugger pin code: 199-876-771

b) `(superset) [root@edeppreapp01 superset-0.15.4]# ./bin/python ./bin/superset runserver -d -r True -p 8888`
/root/superset-0.15.4/lib/python2.7/site-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.
  warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "
2017-02-10 10:38:42,739:INFO:flask_appbuilder.base:Registering class MyIndexView on menu 
2017-02-10 10:38:42,744:INFO:flask_appbuilder.base:Registering class UtilView on menu 
// ...
2017-02-10 10:38:44,748:INFO:flask_appbuilder.base:Registering class Superset on menu 
2017-02-10 10:38:44,832:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-10 10:38:44,871:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-10 10:38:44,962:INFO:werkzeug: * Running on http://0.0.0.0:8888/ (Press CTRL+C to quit)
2017-02-10 10:38:44,963:INFO:werkzeug: * Restarting with stat
// ...
2017-02-10 10:38:47,743:INFO:flask_appbuilder.base:Registering class Superset on menu 
2017-02-10 10:38:47,831:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-10 10:38:47,869:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-10 10:38:47,959:WARNING:werkzeug: * Debugger is active!
2017-02-10 10:38:47,960:INFO:werkzeug: * Debugger pin code: 199-876-771

c) `(superset) [root@edeppreapp01 superset-0.15.4]# ./bin/python ./bin/superset runserver -d -r False -p 8888`
/root/superset-0.15.4/lib/python2.7/site-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.
  warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "
2017-02-10 10:40:13,928:INFO:flask_appbuilder.base:Registering class MyIndexView on menu 
2017-02-10 10:40:13,933:INFO:flask_appbuilder.base:Registering class UtilView on menu 
// ...
2017-02-10 10:40:16,036:INFO:flask_appbuilder.base:Registering class CssTemplateModelView on menu CSS Templates
2017-02-10 10:40:16,074:INFO:flask_appbuilder.base:Registering class CssTemplateAsyncModelView on menu 
2017-02-10 10:40:16,171:INFO:werkzeug: * Running on http://0.0.0.0:8888/ (Press CTRL+C to quit)

File: superset/cli.py [hunk 1]
```diff
@@ -31,6 +31,9 @@ def init():
     '-d', '--debug', action='store_true',
     help="Start the web server in debug mode")
 @manager.option(
```
Comment (mistercrunch) [inline superset/cli.py]: It's not well documented but it's similar to argparse, so you can ` action='store_true'` as in
```
@manager.option(
    '-n', '--no-reload', action='store_true', dest='no_reload',
    help="Don't use the reloader in debug mode")
```

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset/cli.py (hunk 1 above)
Comment (mistercrunch) [inline superset/cli.py]: then no_reload has to be in the function args and will be bool

Comment (asdf2014) [inline superset/cli.py]: Great! Thx.

**Comment by asdf2014** [comment]
I changed it into `action='store_false'` to look more reasonable :-)

**Pull request merged**